### PR TITLE
[REF] mail: flatten store data sent on network (part 1)

### PR DESCRIPTION
**Pull request by seb-odoo**
This will reduce data transferred to the client and processed in JS when the same member was appearing multiple times.

This also makes more clear what is transferred by avoiding nested values.

Note: this is a first step, many format methods still have to be converted, in particular the one from member, which will then allow converting the one from partner, which will significantly reduce duplicate data send for the current partner (one duplicate per open channel).

Formatting code eventually becomes simpler too, by always adding data to the store rather than updating pre-existing dict manually.

Part of task-3605717

https://github.com/odoo/enterprise/pull/64700

### CODE REVIEW

File: addons/mail/tools/discuss.py [hunk 1]
```diff
@@ -39,49 +41,67 @@ def get_sfu_key(env) -> str | None:
...
+            values = [values]
+        assert isinstance(values, list), f"expected list for {model_name}: {values}"
+        for vals in values:
+            assert isinstance(vals, dict), f"expected dict for {model_name}: {vals}"
+            for i in ids:
+                assert vals.get(i), f"missing id {i} in {model_name}: {vals}"
+            match = filter(lambda record: all(record[i] == vals[i] for i in ids), self.data[model_name])
+            if record := next(match, None):
+                record.update(vals)
+            else:
```
Comment (alexkuhn) [inline addons/mail/tools/discuss.py:102]: `store.add()` twice on same record whose `many` field was `["ADD", {}]` will no combine the list of commands and instead just the latter list of commands will be present.

Not a huge problem for now, but just pointing out a limitation to be aware of.

**Review by alexkuhn — APPROVED** [review]
LGTM
@robodoo delegate+

File: addons/mail/tools/discuss.py [hunk 2]
```diff
@@ -39,49 +41,67 @@ def get_sfu_key(env) -> str | None:
...
+            return self
+        ids = ids_by_model[model_name]
+        # handle singleton: update in place
+        if len(ids) == 0:
+            assert isinstance(values, dict), f"expected dict for singleton {model_name}: {values}"
+            if not model_name in self.data:
+                self.data[model_name] = {}
+            self.data[model_name].update(values)
+            return self
+        # handle (multi) id(s): add or update existing
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/mail/tools/discuss.py:89]: Not a big deal, if you're ready to approve it, I'd rather update the comment in a future update and get this merged ASAP, as I already have follow up commits depending on this and it's not the easiest to keep out of conflicts.

**Comment by alexkuhn** [comment]
@robodoo override=ci/security r+ rebase-ff
`Markup` on explicit string, `sendone` on safe channel (discuss channel and partner)

**Pull request closed without merging**